ci-operator/step-registry/platform-external/pre/conf/manifests/platform-external-pre-conf-manifests-commands.sh

```diff
@@ -0,0 +1,138 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+# TEMP until figure out issues in deployment when declaring release:initial in workflow
+if [[ -n "${PLATFORM_EXTERNAL_OVERRIDE_RELEASE-}" ]]; then
+  export OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE="${PLATFORM_EXTERNAL_OVERRIDE_RELEASE}"
+fi
+echo "Using release image ${OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE}"
+
+STEP_WORKDIR=${STEP_WORKDIR:-/tmp}
+INSTALL_DIR=${STEP_WORKDIR}/install-dir
+mkdir -vp "${INSTALL_DIR}"
+
+source "${SHARED_DIR}/init-fn.sh" || true
+install_butane
+
+log "Copying to install dir"
+cp -vp "${SHARED_DIR}"/install-config.yaml "${INSTALL_DIR}"/install-config.yaml
+
+log "Creating manifests"
+openshift-install create manifests --dir "${INSTALL_DIR}"
+
+log "\n
+#
+# << Manifest customization >>
+#"
+
+#
+# MachineConfig for kubelet providerId
+#
+function create_machineconfig_kubelet() {
+    local node_role=$1
+    # shellcheck disable=SC1039
+    cat << EOF > "$STEP_WORKDIR/mc-kubelet-${node_role}.bu"
+variant: openshift
+version: 4.13.0
+metadata:
+  name: 00-$node_role-kubelet-providerid
+  labels:
+    machineconfiguration.openshift.io/role: $node_role
+storage:
+  files:
+  - mode: 0755
+    path: "/usr/local/bin/kubelet-providerid"
+    contents:
+      inline: |
+        #!/bin/bash
+        set -e -o pipefail
+        NODECONF=/etc/systemd/system/kubelet.service.d/20-providerid.conf
+        if [ -e "\${NODECONF}" ]; then
+            echo "Not replacing existing \${NODECONF}"
+            exit 0
+        fi
+
+        PROVIDER_ID=${PROVIDER_ID_COMMAND}
```

Comment:
Can we add the command to discover the providerId as a field in install-config.yaml, making installer create this MachineConfig when platform is External, and CCM is enabled?